reactioncommerce/example-storefront
# More Docker fixes and improvements

Impact: **minor**
Type: **chore**

## Changes
- Production Docker image is now built with fewer dependencies installed and in a way that ensures it will run by default in production mode (not rebuilding pages as they are accessed)
- The new public image `reactioncommerce/node-dev:10.16.3-v2` is used when running locally using docker-compose. `Dockerfile.dev` is no longer needed and is removed.
- Support files and scripts that are no longer used are removed.
- The CircleCI config is updated to match recent changes made in the API repo. Linting and tests do not run in the Docker image. It is built and pushed later, only if checks pass. The standard scripts from the `ci-scripts` package are used.

## Breaking changes
None

## Testing
- Verify you can run successfully with `docker-compose up`
- Stop the dev app with `docker-compose down`, and then do `docker build -t example-storefront .` to make a production build on your machine. Then run it with
    - Clear your `.env` file - run `bin/setup` to start fresh
    - `docker run -it --name storefront -p 4000:4000 --env-file .env --network api.reaction.localhost example-storefront`
    - Verify you do NOT see a "Running NextJS server in development mode" message in the logs when the server starts
    - Verify the website works at localhost:4000. Pages should load very quickly compared with running in dev mode.
- Similar to the above test, but run the image that was created by CircleCI for this branch:
    - `docker run -it --name storefront -p 4000:4000 --env-file .env --network api.reaction.localhost reactioncommerce/example-storefront:chore-use-node-dev-docker-image`
    - It will download the image from DockerHub automatically.
